bm.el

```diff
@@ -555,22 +575,55 @@ before bm is loaded.")
   (interactive)
   (customize-group 'bm))
 
+(defun bm--suggest-lisp (&optional bm)
+  "Suggestion function for lisp mode bookmark annotations."
+    (when bm
+      (goto-char (overlay-start bm)))
+    (end-of-line)
+    (cond
+     ((re-search-backward "^(\\(def[^ ]+ [^ ]+\\)" nil t)
+       (buffer-substring (match-beginning 1) (match-end 1)))
+     ((re-search-backward "^;;;\s+\\([^\s].*$\\)" nil t)
+       (buffer-substring (match-beginning 1) (match-end 1)))
+     (t "")))
+
+(defun bm--suggest-org (&optional bm)
+  "Suggestion function for org mode bookmark annotations."
+    (when bm
+      (goto-char (overlay-start bm)))
+    (end-of-line)
+    (if (re-search-backward org-heading-regexp nil t)
+      (buffer-substring (match-beginning 0) (match-end 0))
+     ""))
+
+(defun bm--suggest-annotation (&optional bm)
```

Comment:
This should be an extension point rather than a specific function. Then users can add the specifics themselves.